### ISSUE TITLE
Gradient of maximum score with respect to input image is 'None'

### ISSUE THREAD

**Issue by RMalikM** [ZF1044404254/MMAL-Net#8]
I am trying to compute the saliency maps for MMAL-Net. I am following [this](https://medium.com/datadriveninvestor/visualizing-neural-networks-using-saliency-maps-in-pytorch-289d8e244ab4) blog. I have trained MMAL-Net on my custom data. To compute the saliency map we first forward an image through the network and compute the score. Then we need to compute gradients of maximum score with respect to each pixel of the input image. This is done using the [backward()](https://pytorch.org/docs/master/_modules/torch/autograd.html#grad) function from `torch.autograd()`. In my case, when I use `backward()` on maximum score its gradients are None. 

While the method in the [blog](https://medium.com/datadriveninvestor/visualizing-neural-networks-using-saliency-maps-in-pytorch-289d8e244ab4) works other model that are available in torch but for MMAL-Net gradients are None. 

Is there any suggestions how to fix it or am I missing something?
